### ISSUE TITLE
[Feature Request] Multi select in tree view

### ISSUE THREAD

**Issue by TheRedD0t** [getinsomnia/insomnia#493]
It would be good to be able to move/duplicate/delete/etc. multiple request and folders in the tree view at once using combinations such as:

**Mouse**
    <kbd>shift</kbd>+`left click`: Select all items from the currently focused item to the clicked item
    <kbd>ctrl</kbd>+`left click`: Add the clicked item to the selection

**Keyboard**
    <kbd>shift</kbd>+<kbd>↑</kbd>: Add the item above the focused item to the selection (or reduce the selection)
    <kbd>shift</kbd>+<kbd>↓</kbd>: Add the item below the focused item to the selection (or reduce the selection)
    <kbd>ctrl</kbd>+<kbd>↑</kbd>: Move the focus up without selecting the item
   <kbd>ctrl</kbd>+<kbd>↓</kbd>: Move the focus down without selecting the item
    <kbd>ctrl</kbd>+<kbd>space</kbd>: Add/remove the focused item from the selection

**Comment by gschier** [comment]
I don't think I have time to work on this but I would love to provide guidance if you wanted to put a PR together for it.

**Comment by gschier** [comment]
I should point out that this is not particularly beginner friendly. It would require quite a bit of knowledge of React, JS, as well as some understanding of the app's architecture.

**Comment by TheRedD0t** [comment]
Sorry, but I'm not familiar with React and I don't have enough time to get into it right now. :\\
But that this feature is something you shouldn't need that often anyway, so no hurry.

**Comment by gschier** [comment]
That's okay, it was worth a shot 😄
I'll leave this issue open for now in case someone else wants to take it on.

**Comment by kikeh** [comment]
Any updates on this?

This is a very basic feature, I can't believe it is not implemented from the beginning 😢 It is true that it's not something that you do every day, but it is quite annoying whenever you have a lot of requests and want to rearrange them in groups. I hope it gets implemented someday!

**Comment by sgtlambda** [comment]
Is anyone working on this? If not I'm willing to look into it

**Comment by Roman-Jordan** [comment]
I am totally going to try to do this. :) I love this app and will be eat sleep and breathing its source for the next week or so. :)

**Comment by benjivm** [comment]
This is a pretty basic feature that is desperately needed for people like me who work with tons of different requests, any updates?

**Comment by LukePeltier** [comment]
This would be awesome

**Comment by DavidExtern** [comment]
What is the status ? When we are importing a big swagger, it's take too long to delete one by one

**Comment by danielbom** [comment]
https://jmespath.org/tutorial.html#multiselect please!

**Comment by dbtrnl** [comment]
Yeah, same here. Just imported the Github API (~735 calls) on the wrong place and have to move one by one to the right folder.
Pretty annoying that such a basic UI feature is missing.

**Comment by BTripp1986** [comment]
.

**Comment by Aleppe** [comment]
No update on this?

**Comment by Anutrix** [comment]
Any updates on this?
This is blocking simple things like moving a list of requests into a new folder unless there's an alternate way to do this.
Often when importing openapispec file may put all large number of request openly.

**Comment by RukshanJS** [comment]
I can work on this with some guidance @gschier I Have decent knowledge on the tech stack.

**Comment by j-q-in-berlin** [comment]
Any news on this? :)

**Comment by Crustack** [comment]
> I can work on this with some guidance [@gschier](https://github.com/gschier) I Have decent knowledge on the tech stack.

Hey, did you have time to work on this?
This would be so convenient, its very annoying as many people described here. Even if you only want to move multiple requests into a subfolder, this would help A LOT

**Comment by emperador-ming** [comment]
It would be a great feature!
